src/app_ui/sign/widgets.rs

```diff
@@ -38,18 +43,7 @@ pub fn display_receiving() {
     screen_update();
 }
 
-pub fn delegate_error_screen() {
-    clear_screen();
-
-    // Add icon and text to match the C SDK equivalent.
-    if cfg!(target_os = "nanos") {
-        "Sign delegate action".place(Location::Custom(2), Layout::Centered, true);
-        "not supported...".place(Location::Custom(14), Layout::Centered, true);
-    } else {
-        WARNING.draw(57, 10);
-        "Sign delegate action".place(Location::Custom(28), Layout::Centered, true);
-        "not supported...".place(Location::Custom(42), Layout::Centered, true);
-    }
-
-    screen_update();
+#[cfg(any(target_os = "stax", target_os = "flex"))]
+pub fn display_receiving() {
+    // TODO: implement loader indicator for stax and flex
```

Comment:
What kind of API do you need here ? Something that just displays a spinner with some text ? But when should it return ?

Reply:
Yes, we would need some loading indicator. After showing it, we dive into receiving a stream of data and deserializing it on the go.

https://github.com/LedgerHQ/app-near-new/blob/dc6e6a2a9d68186a82cdcb964679528ff792f9a9/src/handlers/sign_tx.rs#L56

Reply:
So it needs to be a non-blocking call, right ?

I'm asking so we can improve the SDK, not to annoy you.

Reply:
Yes, it needs to be non-blocking

Reply:
Ok thx.